plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -5611,4 +5612,125 @@ public boolean attachConfigDriveToMigratedVm(Connection conn, String vmName, Str
 
     }
 
+    /**
+     * Get Diagnostics Data API
+     * Copy zip file from system vm and copy file directly to secondary storage
+     * @param conn
+     * @param cmd
+     * @return
+     */
+    public Answer copyDiagnisticsFileToSecondaryStorage(Connection conn, CopyToSecondaryStorageCommand cmd){
+        String fileInSecondaryStorage;
+        String secondaryStorageUrl = cmd.getSecondaryStorageUrl();
+        String vmIP = cmd.getSystemVmIp();
+        //fileName = /root/<fileName>
+        String fileInVR = cmd.getFileName();
+
+        Answer answer = null;
+        String localDir = null;
+
+        // Mount Secondary storage and create diagnostics dir
+        String secondaryStorageMountPath = null;
+        try {
+            URI uri = new URI(secondaryStorageUrl);
+            secondaryStorageMountPath = uri.getHost() + ":" + uri.getPath();
+            String diagnosticsFileDir = "diagnostics_data";
+            if (!createSecondaryStorageFolder(conn, secondaryStorageMountPath, diagnosticsFileDir)){
+                String msg = "Failed to create diagnostics folder in secondary storage";
+                s_logger.error(msg);
+               answer = new Answer(cmd, false, msg);
```

Comment:
At this point wouldn't you want to return this answer? As it is now it would continue trying to mount secondary storage

Reply:
I actually don’t need that check, I have delegated the creation of the diagnostics_data directory to the scp command similar to how I do it for kvm and VMware